### PR TITLE
`renderSubtreeintoContainer` should export instance of component.

### PR DESCRIPTION
**Pull request by windyGex**
@developit

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -453,9 +453,6 @@ function bindAll(ctx) {
 
 
 function callMethod(ctx, m, args) {
-	if (typeof m==='string') {
```
Comment (developit) [inline src/index.js]: Is this change necessary for the PR?  If not please remove it and submit as another PR - at first glance, it breaks libraries that reassign to `this.render` after instantiation.

File: src/index.js [hunk 2]
```diff
@@ -168,7 +168,7 @@ function renderSubtreeIntoContainer(parentComponent, vnode, container, callback)
 	let wrap = h(ContextProvider, { context: parentComponent.context }, vnode);
 	let c = render(wrap, container);
 	if (callback) callback(c);
-	return c;
+	return c._component || c.base;
```
Comment (developit) [inline src/index.js]: Strange - it seems like this should have already been taken care of by [L154](https://github.com/developit/preact-compat/pull/295/files#diff-1fdf421c05c1140f6d71444ea2b27638L154), shouldn't it?  I wonder why it wasn't working...

Comment (windyGex) [inline src/index.js]: Yes, it's work. But the returned value is instance of `ContextProvider`, not passed vnode.

File: src/index.js (hunk 1 above)
Comment (windyGex) [inline src/index.js]: Ok, I will remove this change and reopen another PR.

File: src/index.js (hunk 2 above)
Comment (developit) [inline src/index.js]: ooooh - wow, yeah that's a very good catch. We'll just want to have this happen only for ContextProvider.
